lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -878,17 +876,19 @@ def _bulk_dispatch(
             auditing["status"] = AuditStatus.FAILURE
             auditing["reason"] = AuditReason.INTERNAL
 
-            # The DELETE operator removes the "sync" context to signal that
-            # the operations table rows no longer exist, so check.
+            # The DELETE API removes the "sync" context to signal that the
+            # operations table rows no longer exist, so check; but if it's
+            # still set then our `sync` local is valid.
             if context.get("sync"):
-                context["sync"].error(dataset, f"update error {e}")
+                sync.error(dataset, f"update error {e}")
```

Comment:
`bool(context.get("sync"))` strikes me as heavier-weight than `"sync" in context`, since the former actually has to do the lookup, evaluate it, and return the default value in the `False` case and then convert it to a boolean...but OK.

(However, if you're inclined to change it, do change line 889, too.  🙂)